src/python/pants/build_graph/mirrored_target_option_mixin.py

```diff
@@ -0,0 +1,92 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from abc import abstractmethod, abstractproperty
+
+from future.utils import text_type
+
+from pants.util.memo import memoized_property
+from pants.util.meta import AbstractClass
+from pants.util.objects import datatype
+
+
+class MirroredTargetOptionDeclaration(AbstractClass):
+  """An interface for operations to perform on an option which may also be set on a target."""
+
+  @abstractproperty
+  def is_flagged(self):
+    """Whether the option was specified on the command line."""
+
+  @abstractmethod
+  def extract_target_value(self, target):
+    """Get the value of this option from target. May return None if not set on the target."""
```

Comment:
Rather than "May", this should be "Must", I think?

Reply:
Fixed this doc to make it clear that a `None` return value is interpreted as "not set".